### PR TITLE
Add unfilterable `wp-` prefix to sitemaps

### LINKED ISSUE #128: prefix sitemap(-*).xml to wp-sitemap(-*).xml
### Description

To prevent collisions with existing plugins, we discussed to rename the `sitemap(-*).xml` to `wp-sitemap(-*).xml`.

The file-name(s) should be also filterable, to help other plugins, that use the new core feature, to be consistent.

### PR DESCRIPTION
**Pull request by swissspidy**
### Issue Number
Fixes #128.

### Description
Adds unfilterable `wp-` prefix to sitemaps

### Type of change
Please select the relevant options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Steps to test
Describe the tests required to verify your changes.
Provide instructions so the PR Tester can check functionality and also list any relevant details and / or dependencies required for your tests.

### Acceptance criteria
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [x] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added test instructions that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by joemcgill — APPROVED** [review]
👍

File: inc/class-core-sitemaps.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ public function register_rewrites() {
 		add_rewrite_tag( '%sub_type%', '([^?]+)' );
 
 		// Register index route.
-		add_rewrite_rule( '^sitemap\.xml$', 'index.php?sitemap=index', 'top' );
+		add_rewrite_rule( '^wp-sitemap\.xml$', 'index.php?sitemap=index', 'top' );
```
Comment (pfefferle) [inline inc/class-core-sitemaps.php:121]: `index.php?sitemap=index` could also collide with other plugins, why not also use the `wp-` prefix: `index.php?wp-sitemap=index` ?

Comment (swissspidy) [reply inc/class-core-sitemaps.php:121]: Mind creating an issue for that?

Comment (pfefferle) [reply inc/class-core-sitemaps.php:121]: sure!

**Pull request merged**